### PR TITLE
add declaration base component

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

add declaration base component

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Tested locally

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: SPDBCSC-229 part 1.

### CODE REVIEW

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.declareTitle {
+  width: 100%;
+  height: 36px;
+  background: inherit;
+  background-color: rgba(56, 89, 138, 1);
+  box-sizing: border-box;
+  border-width: 3px;
+  border-style: solid;
+  -moz-box-shadow: none;
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/base/declaration/Declaration.css]: do we need to provide these properties if they're just being set to `none`?

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  -webkit-box-shadow: none;
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/base/declaration/Declaration.css]: same question here

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  box-shadow: none;
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/base/declaration/Declaration.css]: and here

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  color: #333333;
+  text-align: left;
+  line-height: 20px;
+}
+
+.declareSection {
+  padding-left: 20px;
+  padding-right: 20px;
+  padding-top: 20px;
+  padding-bottom: 0px;
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/base/declaration/Declaration.css]: does this need to be provided or is default automatically set to 0px?

File: src/ecrc-frontend/src/components/base/declaration/Declaration.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          of facts herein may lead to a denial of a cannabis workers
+          registration. I am also aware that later discovery of an omission or
+          misrepresentation may be grounds for any finding of suitability to be
+          suspended or revoked.
+        </span>
+      </section>
+
+      <section className="declareSection" style={{ width: "475px" }}>
+        <TextInput
+          textInput={{ ...textInput }}
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/base/declaration/Declaration.js]: why not just
```suggestion
          textInput={textInput}
```

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply src/ecrc-frontend/src/components/base/declaration/Declaration.css]: ok. will remove it.

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css (hunk 4 above)
Comment (peggy-gov-bc-ca) [reply src/ecrc-frontend/src/components/base/declaration/Declaration.css]: removed.

File: src/ecrc-frontend/src/components/base/declaration/Declaration.js (hunk 5 above)
Comment (peggy-gov-bc-ca) [reply src/ecrc-frontend/src/components/base/declaration/Declaration.js]: ok

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  width: 100%;
+  height: 36px;
+  background: inherit;
+  background-color: rgba(56, 89, 138, 1);
+  box-sizing: border-box;
+  border-width: 3px;
+  border-style: solid;
+  font-family: "myriad-pro", "sans serif";
+  font-weight: 400;
+  font-style: normal;
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/base/declaration/Declaration.css]: is normal the default font-style? in which case we may not need to provide this value

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css [hunk 7]
```diff
@@ -0,0 +1,29 @@
+.declareTitle {
+  width: 100%;
+  height: 36px;
+  background: inherit;
+  background-color: rgba(56, 89, 138, 1);
+  box-sizing: border-box;
+  border-width: 3px;
+  border-style: solid;
+  font-family: "myriad-pro", "sans serif";
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/base/declaration/Declaration.css]: no custom fonts, we are using bcsans throughout the application

Comment (peggy-gov-bc-ca) [reply src/ecrc-frontend/src/components/base/declaration/Declaration.css]: ok.

File: src/ecrc-frontend/src/components/base/declaration/Declaration.css (hunk 6 above)
Comment (peggy-gov-bc-ca) [reply src/ecrc-frontend/src/components/base/declaration/Declaration.css]: agree

**Pull request merged**